backend/src/main/java/com/festago/festival/application/FestivalSearchV1QueryService.java

```diff
@@ -15,16 +24,64 @@
 @RequiredArgsConstructor
 public class FestivalSearchV1QueryService {
 
-    private static final Pattern SCHOOL_PATTERN = Pattern.compile(".*대(학교)?$");
-
     private final ArtistFestivalSearchV1QueryDslRepository artistFestivalSearchV1QueryDslRepository;
     private final SchoolFestivalSearchV1QueryDslRepository schoolFestivalSearchV1QueryDslRepository;
 
     public List<FestivalSearchV1Response> search(String keyword) {
-        Matcher schoolMatcher = SCHOOL_PATTERN.matcher(keyword);
-        if (schoolMatcher.matches()) {
-            return schoolFestivalSearchV1QueryDslRepository.executeSearch(keyword);
+        return sortFestival(findFestivals(keyword));
+    }
+
+    private List<FestivalSearchV1Response> findFestivals(String keyword) {
+        if (artistFestivalSearchV1QueryDslRepository.existsByName(keyword)) {
+            return artistFestivalSearchV1QueryDslRepository.executeSearch(keyword);
+        }
+        return schoolFestivalSearchV1QueryDslRepository.executeSearch(keyword);
+    }
+
+    private List<FestivalSearchV1Response> sortFestival(List<FestivalSearchV1Response> festivals) {
+        Map<FestivalFilter, List<FestivalSearchV1Response>> festivalByStatus = divideByStatus(festivals);
+        List<FestivalSearchV1Response> result = new ArrayList<>();
+        for (FestivalFilter festivalFilter : determineFestivalOrder()) {
+            List<FestivalSearchV1Response> festivalsByFilter = festivalByStatus.getOrDefault(festivalFilter,
+                new ArrayList<>());
```

Comment:
반환되는 리스트에 변경이 없다면 `Collections.emptyList()`를 사용해도 좋을 것 같네요!
```suggestion
            List<FestivalSearchV1Response> festivalsByFilter = festivalByStatus.getOrDefault(festivalFilter,
                Collections.emptyList());
```